### DISCUSSION TITLE
Tried to deploly voskjshttp.js to heroku getting Error

### DISCUSSION THREAD

**Discussion by believeitcode** [solyarisoftware/voskJs#5] [Q&A]
Heroku log: 
ERROR (VoskAPI:ReadConfigFile():parse-options.cc:462) Cannot open config file: ./models/model-en/mfcc.conf
terminate called after throwing an instance of 'kaldi::KaldiFatalError'
what(): kaldi::KaldiFatalError

Project objective
Hi solyaris , thanks for the hard work and documentation , this is my first time using voskjs. I'm current doing my graduation project using misty robot would like to give credit to you in my paper , i would like to deploly the nodejs http server file with the model onto Heroku . From misty robot i will send a external GET request with the recorded voice wave file to the server for the objective of speech to text feature instead of using Google / Azure STT . Later on i will be doing running API for NLP for Question and awnswer.

![heroku crash](https://user-images.githubusercontent.com/40289508/118994084-9f242800-b9b8-11eb-8ed1-97520b0dfd87.png)
  
Problem :
Currently my test of using node voskjshttp.js --model=./models/model-en to start up the sever is working i using the small language english model and doing a GET request from postman http://localhost:3000/transcript?speech=C:\Users\PCUser\OneDrive\Desktop\voskJs-master\voskJs-master\audio\2830-3980-0043.wav is working alright but with heroku i got a error as shown from the log above when deployed sucessfully. 

I would appreciate if someone could assist me on finding out the problem. Thanks

Below is the repository with minimal code changes on port to deploy on heroku and also changed the name of the model folder to model-en 
https://github.com/believeitcode/voskjs-misty

**Comment by solyarisoftware** [comment]
Hi Tay

It seems to me that the issue is caused by the fact VoskJs is NOT able to open the Model directory `./models/model-en`.

Are you loaded the model directory files on the Heroku dyno? 

BTW, I'm not familiar with heroku but please note the heroku file system is "ephemeral":
https://help.heroku.com/K1PPS2WM/why-are-my-file-uploads-missing-deleted

so programs probably can NOT write files (if i well remenber).

- If you want/need to use a computer on the cloud Why don't you consider to use a virtual machine (for basic needs it costs few dollars/months)

- Or, because you are running a robot app for https://www.mistyrobotics.com/ , maybe any on-premise server (as a small linux server) could be a better choice.


Last but not least, please note VoskJs is in a very draft stage and APIs could change.

**Reply by believeitcode** [reply]
Thanks for the reply Giorgio Robino. Yeap , i loaded model directory file on dyno and thanks for pointed out about heroku file system "ephemeral" .  This misty robot is  very new to me , i just only had 2 week of experience with it . Most of the example tutorial provided by developer team was using 3rd party service with Azure and Google cloud build on functionality with the robot personally trying not to use them if possible. The robot is running on it own SDK in javascript include .NET or using REST , which i feel it will be heavily depend on APIs. 

One of the  misty developer shared with me that they had implemented Sphnix and DeepSpeech on the robot internally but did not share much in-depth detail about it , i was just trying out using a vosk on it . 

Btw thanks for documenting about vosk ,i agree is a very fresh library . But you had documented it well than the offical website it self thanks alot  !

**Comment by solyarisoftware** [comment]
I'm not familiar with misty robot but probably you have many options, depending to which architecture you want to build.

Do you really need a server ? 
Probably if want to make a prototype you have a single robot. In that case you don't need a server and you can use direct Vosk / VoskJs API in your program. See [example](https://github.com/solyarisoftware/voskJs/blob/master/examples/simple.js)

Is it ok to use am ASR on the cloud?
For an academic project i'd suggest this free service: https://wit.ai/docs/http/20200513/#post__speech_link

Do you want a low latency offline ASR? 
Use VoskJs!

Do you want to experiment with Coqui STT (successor of DeepSpeech), see my project:
https://github.com/solyarisoftware/CoquiSTTJs

Stars on github are appreciated :)
